syntax = "proto3";

package deepface.api.proto;

import "common.proto";

message VerifyRequest {
    bytes image1 = 1;
    bytes image2 = 2;
    optional deepface.api.proto.Models model_name = 3;
    optional deepface.api.proto.Detectors detector_backend = 4;
    optional deepface.api.proto.DistanceMetrics distance_metric = 5;
    optional bool enforce_detection = 6;
    optional bool align = 7;
    optional bool anti_spoofing = 8;
}

message VerifyResponse {

    message FacialAreas {
        deepface.api.proto.FacialArea img1 = 1;
        deepface.api.proto.FacialArea img2 = 2;
    }

    bool verified = 1;
    deepface.api.proto.Detectors detector_backend = 2;
    deepface.api.proto.Models model = 3;
    deepface.api.proto.DistanceMetrics similarity_metric = 4;
    FacialAreas facial_areas = 5;
    double distance = 6;
    double threshold = 7;
    double time = 8;
}

service VerifyService {
    rpc Verify(VerifyRequest) returns (VerifyResponse);
}